homotopy-core/src/diagram.rs

```diff
@@ -405,6 +446,17 @@ impl DiagramN {
         Self::new(source.into(), cospans)
     }
 
+    #[must_use]
+    pub fn replace(&self, s: Generator, t: Generator, a: Generator) -> Self {
+        let source = self.source().replace(s, t, a);
+        let cospans: Vec<_> = self
+            .cospans()
+            .iter()
+            .map(|c| c.map(|r| r.replace(s, t, a)))
+            .collect();
+        Self::new(source, cospans)
```

Comment:
I wonder if we should have a `map` function on `DiagramN` since this pattern is becoming quite common.